app/components/CommentForm/index.js

```diff
@@ -1,153 +1,122 @@
-        </div>
+  return (
+    <LegoFinalForm
+      initialValues={{
+        text: EDITOR_EMPTY,
+      }}
+      validate={validate}
+      onSubmit={({ text }) => {
+        return dispatch(
+          addComment({
+            contentTarget,
+            text,
+            parent,
+          })
+        );
+      }}
+    >
+      {({ handleSubmit, pristine, submitting, form }) => {
+        const textValue = form.getFieldState('text')?.value;
+        const fieldActive = form.getFieldState('text')?.active;
+        const isOpen = fieldActive || (textValue && textValue !== EDITOR_EMPTY);
 
-        <div
-          className={cx(styles.fields, isOpen && styles.activeFields)}
-          onMouseOver={this.enableForm}
-          onScroll={this.enableForm}
-          onPointerDown={this.enableForm}
-        >
-          {this.state.disabled ? (
-            <DisplayContent
-              id="comment-text"
-              className={styles.text}
-              content=""
-              placeholder="Skriv en kommentar"
-            />
-          ) : (
-            <Field
-              autoFocus={autoFocus}
-              name="text"
-              placeholder="Skriv en kommentar"
-              component={EditorField}
-              initialized={initialized}
-              simple
-            />
-          )}
+        return (
+          <form
+            onSubmit={handleSubmit}
+            className={cx(className, isOpen && styles.activeForm)}
+          >
+            <div className={styles.header}>
+              <ProfilePicture size={40} user={user} />
 
-          {isOpen && (
-            <Button
-              submit
-              disabled={pristine || submitting}
-              className={styles.submit}
-            >
-              {submitText}
-            </Button>
-          )}
-        </div>
-      </form>
-    );
-  }
-}
+              {isOpen && <div className={styles.author}>{user.fullName}</div>}
+            </div>
+            <div className={cx(styles.fields, isOpen && styles.activeFields)}>
+              {disabled ? (
+                <DisplayContent
+                  id="comment-text"
+                  className={styles.text}
+                  content=""
+                  placeholder="Skriv en kommentar"
+                />
+              ) : (
+                <Field
+                  autoFocus={autoFocus}
+                  name="text"
+                  placeholder="Skriv en kommentar"
+                  component={EditorField.AutoInitialized}
+                  simple
+                />
+              )}
 
-function mapStateToProps(state, props) {
-  const meta = getFormMeta(props.form)(state);
-  const values = getFormValues(props.form)(state);
-  return {
-    isOpen:
-      meta &&
-      meta.text &&
-      (meta.text.active || (values && values.text !== EMPTY_STATE)),
-  };
-}
+              {isOpen && (
+                <Button
+                  submit
+                  disabled={pristine || submitting}
```

Comment:
Why are we not doing `disabled={disabled || pristine || submitting}`? Also, the button is never disabled again after being "activated", which might appear like a bug to a user. Kinda out of scope of what you're trying to accomplish in this PR though, but I wanted to shed some light on it.